tensor_genn/algorithms/relu_ann.py

```diff
@@ -42,23 +42,63 @@ def create_weight_matrices(self, tf_model):
                     gw_inds.append(None)
                     gw_vals.append(syn_weights.flatten())
 
-                # Prepare weight matrices for Conv2D
                 elif isinstance(layer,tf.keras.layers.Conv2D):
+                    # Prepare weight matrices for Conv2D
+                    '''
+                    Indexing of neurons is done in the order: channels -> width -> height
+                    Assign weights for every kw*ic block in the input and map to output neuron
+                    corresponding to that convolution operation.
+                    '''
+
                     kw,kh = layer.kernel_size
                     sw,sh = layer.strides 
                     ih,iw,ic = layer.input_shape[1:]
                     oh,ow,oc = layer.output_shape[1:]
-        
-                    for n in range(int(n_units[j])): # output unit index/conv number
-                        for k in range(kh): # over kernel height
-                            '''
-                            Indexing of neurons is done in the order: channels -> width -> height
-                            Assign weights for every kw*ic block in the input and map to output neuron
-                            corresponding to that convolution operation.
-                            '''
-                            syn_weights[((n//oc)%ow)*ic*sw + ((n//oc)//ow)*ic*iw*sh + k*ic*iw:
-                                        ((n//oc)%ow)*ic*sw + ((n//oc)//ow)*ic*iw*sh + k*ic*iw + kw*ic,
-                                        n] = tf_weights[i-1][k,:,:,n%oc].flatten()
+
+                    if layer.padding == 'valid':
```

Comment:
It might be nice to note in a comment here that 'valid' infact means no padding!